src/gui/preferences.c

```diff
@@ -2035,6 +2066,17 @@ static void edit_preset(GtkTreeView *tree, const gint rowid, const gchar *name,
     gtk_entry_set_text(g->lens, (const char *)sqlite3_column_text(stmt, 3));
     gtk_spin_button_set_value(g->iso_min, sqlite3_column_double(stmt, 4));
     gtk_spin_button_set_value(g->iso_max, sqlite3_column_double(stmt, 5));
+    const gchar *operation = (const char *) sqlite3_column_text(stmt,15);
+    gboolean islib = FALSE;
+
+    for (const GList * lib_modules = g_list_first(darktable.lib->plugins); lib_modules; lib_modules = g_list_next(lib_modules))
```

Comment:
Please split this out in a separate function.

The `g_list_first` isn't required since `->plugins` points to the start of the list. @ralfbrown has just been cleaning this up all across the codebase :-)